[DOCS] API custom logic example (#12627)
Sylius/Sylius: docs/getting-started-with-sylius/custom-business-logic.rst

|

Or if you are using unified API:

> First You need to pickup new cart:

Reviewer: ```suggestion
First, you need to pickup a new cart:
```

.. code-block:: bash

    curl -X POST "https://master.demo.sylius.com/api/v2/shop/orders" -H "accept: application/ld+json"

With empty body:

> .. code-block:: json

Reviewer 2: I suggest using here `locale` and explain how it works

    {}

> This should return a response with ``tokenValue`` which we would need for next API calls:

Reviewer: ```suggestion
This should return a response with ``tokenValue`` which we would need for the next API calls:
```

.. code-block:: json

    {
        "shippingState": "string",
        "tokenValue": "CART_TOKEN",
        "id": 123,
    }

> Then we need to add product to the cart but first it would be good to have any, You can use this call to retrieve some products:

Reviewer: ```suggestion
Then we need to add a product to the cart but first, it would be good to have any. You can use this call to retrieve some products:
```

.. code-block:: bash

    curl -X GET "https://master.demo.sylius.com/api/v2/shop/products?page=1&itemsPerPage=30" -H  "accept: application/ld+json"

And choose any product variant IRI that you would like to add to your cart:

.. code-block:: bash

    curl --location --request PATCH 'https://127.0.0.1:8000/api/v2/shop/orders/CART_TOKEN/items' -H 'Content-Type: application/merge-patch+json'

> With product variant of choose in body:

Reviewer: ```suggestion
With a chosen product variant in the body:
```

.. code-block:: json

    {
        "productVariant": "/api/v2/shop/product-variants/PRODUCT_VARIANT_IRI",
        "quantity": 1
    }

> This should return a response with the cart which should contain ``shippingTotal``:

Reviewer: ```suggestion
This should return a response with the cart that should contain a ``shippingTotal``:
```

.. code-block:: json

    {
        "taxTotal": 0,
        "shippingTotal": 500,
        "orderPromotionTotal": 0
    }

.. attention::

    API returns costs in decimal numbers that's why in response it is 500 currency unit shipping total (which stands for 5 USD in this case).

> Now let's change the quantity of our product variant. We can do it by once again calling endpoint as above, or by utilising ``changeQuantity`` endpoint:

Reviewer: ```suggestion
Now let's change the quantity of our product variant. We can do it by calling the endpoint above once again, or by utilizing the ``changeQuantity`` endpoint:
```

|

> This customization should also work when using unified API without any extra steps:

Reviewer: ```suggestion
This customization will also work when using unified API without any extra steps:
```

> First, you need to pickup new cart:

Reviewer: ```suggestion
First, you need to pick up a new cart:
```

.. code-block:: bash

    curl -X POST "https://master.demo.sylius.com/api/v2/shop/orders" -H "accept: application/ld+json"

With body:

.. code-block:: json

    {
        "localeCode": "string"
    }

.. note::

>     The ``localeCode`` value is optional in body of pickup cart. This means that if you won't provide it, the default locale from channel will be used.

Reviewer: ```suggestion
    The ``localeCode`` value is optional in the body of cart pickup. This means that if you won't provide it, the default locale from the channel will be used.
```

This should return a response with ``tokenValue`` which we would need for the next API calls:

.. code-block:: json

    {
        "shippingState": "string",
        "tokenValue": "CART_TOKEN",
        "id": 123,
    }

Then we need to add a product to the cart but first, it would be good to have any. You can use this call to retrieve some products:

.. code-block:: bash

    curl -X GET "https://master.demo.sylius.com/api/v2/shop/products?page=1&itemsPerPage=30" -H  "accept: application/ld+json"

And choose any product variant IRI that you would like to add to your cart:

.. code-block:: bash

>     curl --location --request PATCH 'https://127.0.0.1:8000/api/v2/shop/orders/CART_TOKEN/items' -H 'Content-Type: application/merge-patch+json'

Reviewer 2: ```suggestion
    curl --location --request PATCH 'https://master.demo.sylius.com/api/v2/shop/orders/CART_TOKEN/items' -H 'Content-Type: application/merge-patch+json'
```

With a chosen product variant in the body:

.. code-block:: json

    {
        "productVariant": "/api/v2/shop/product-variants/PRODUCT_VARIANT_IRI",
        "quantity": 1
    }

This should return a response with the cart that should contain a ``shippingTotal``:

.. code-block:: json

    {
        "taxTotal": 0,
        "shippingTotal": 500,
        "orderPromotionTotal": 0
    }

.. attention::

    API returns costs in decimal numbers that's why in response it is 500 currency unit shipping total (which stands for 5 USD in this case).

Now let's change the quantity of our product variant. We can do it by calling the endpoint above once again, or by utilizing the ``changeQuantity`` endpoint:

.. code-block:: bash

>     curl --location --request PATCH 'https://127.0.0.1:8000/api/v2/shop/orders/CART_TOKEN/items/ORDER_ITEM_ID' -H 'Content-Type: application/merge-patch+json'

Reviewer 2: ```suggestion
    curl --location --request PATCH 'https://master.demo.sylius.com/api/v2/shop/orders/CART_TOKEN/items/ORDER_ITEM_ID' -H 'Content-Type: application/merge-patch+json'
```

> With new quantity in body:

Reviewer: ```suggestion
With new quantity in the body:
```
